Graphics/Core/include/NovelRT/Graphics/GraphicsCmdList.hpp

```diff
@@ -0,0 +1,166 @@
+#pragma once
+
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#include <NovelRT/Graphics/RGBAColour.hpp>
+#include <NovelRT/Maths/GeoVector2F.h>
+#include <NovelRT/Utilities/Misc.h>
+#include <memory>
+
+namespace NovelRT::Graphics
+{
+    template<typename TBackend> class GraphicsContext;
+    template<typename TBackend> class GraphicsBuffer;
+    template<typename TBackend> class GraphicsTexture;
+    template<typename TBackend> class GraphicsDescriptorSet;
+    template<typename TBackend> class GraphicsRenderPass;
+    template<template<typename TBackend> typename TResource, typename TBackend> class GraphicsResourceMemoryRegion;
+
+    // TODO: MOVE THIS
+    struct ViewportInfo
+    {
+        float x;
+        float y;
+        float width;
+        float height;
+        float minDepth;
+        float maxDepth;
+    };
```

Comment:
Are you still moving this?

Reply:
In a later submission, yes. This PR is just so old however and we still have bugs to fix, I want to get this submitted to unblock fixing some bugs and getting progress on some other tasks.